### PR TITLE
DPC-1230 Add target for running dpc-go-attribution unit tests and linters

### PR DESCRIPTION
**Pull request by SalGutierrez**
### Fixes [DPC-1230](https://jira.cms.gov/browse/DPC-1230)

We've been carefully adding unit tests to our Golang code, but currently do not have a simple, memorable way to run them all at once.  To improve local development, and to support further devops work, we need a make test target for the v2 Attribution service.
### Proposed Changes

- Make a unit_test.sh that runs all unit tests for the v2 Attribution service
- Create a Makefile target for make unit-test that runs unit_test.sh after any test setup, if needed
- Create a Makefile target for make lint that runs golangci-lint and gosec
- Create a Makefile target for make test that invokes make unit-test and make lint

### Change Details
created a `unit_test.sh` script to run unit tests and output results.
created a `unit-test` make target that runs unit tests
created a `lint` make target that runs liners and security scans
created a `test` make target that runs unit-test and lint make targets.

### Security Implications

- [ ] new software dependencies

- [ ] security controls or supporting software altered

- [ ] new data stored or transmitted

- [ ] security checklist is completed for this change

- [ ] requires more information or team discussion to evaluate security implications

- [x] no PHI/PII is affected by this change

### Feedback Requested

Is the Makefile location ok? Or did we want it at the root of dpc-go (or dpc-app).

### CODE REVIEW

File: dpc-go/dpc-attribution/Makefile [hunk 1]
```diff
@@ -0,0 +1,15 @@
+.PHONY: unit-test
+unit-test:
+	@bash ./unit_test.sh
+
+.PHONY: lint
+lint:
+	@echo "-------------- ATTRIBUTION LINTING STARTED --------------"
+	@golangci-lint run && echo "*********** ATTRIBUTION IS LINT FREE!! ***********" || echo "*********** LINTING FAILED!! ***********"
```
Comment (dhgreene) [inline dpc-go/dpc-attribution/Makefile:8]:
```suggestion
	@cd src && golangci-lint run && echo "*********** ATTRIBUTION IS LINT FREE!! ***********" || echo "*********** LINTING FAILED!! ***********"
```

File: dpc-go/dpc-attribution/Makefile [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+	@echo "-------------- SECURITY SCAN STARTED --------------"
+	@cd src && gosec -fmt=json -out=../results.json ./... && cd ..
```
Comment (dhgreene) [inline dpc-go/dpc-attribution/Makefile:10]:
```suggestion
	@cd src && gosec -fmt=json -out=../results.json ./...
```
At least in my shell, a `cd` command does not affect the following line

File: dpc-go/dpc-attribution/Makefile [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+	@echo "*********** SECURITY SCAN RESULTS: ***********"
+	@cat results.json
+
+.PHONY: test
+test:  unit-test lint
```
Comment (dhgreene) [inline dpc-go/dpc-attribution/Makefile:15]: I have a slight preference to run `lint` before `unit-test`.  Does the order matter to you?

Comment (SalGutierrez) [reply dpc-go/dpc-attribution/Makefile]: Nope, I have changed the order for both services.

**Review by dhgreene — APPROVED** [review]
🧪

**Pull request merged**